### PR TITLE
CAP-0056 - Add topics

### PR DESCRIPTION
**Pull request by sisuresh**
Adds a `SCVec` field to `ContractEvent`.

Related thread - https://discord.com/channels/897514728459468821/989220928384806932/1004194314101989469.

### CODE REVIEW

File: core/cap-0056.md [hunk 1]
```diff
@@ -51,12 +51,14 @@ the hash stored in `TransactionResultPairV2`.
...
-func $contract_event(param $val i64) (result i64)
+/// Will save val and topics to a list that will be returned to core so ContractEvent can be
+/// populated and put into TransactionMetaV3.events with type ContractEventType::CONTRACT.
+/// topics is expected to be a SCVec with length <= 4 that cannot contain Vecs, Maps,
+/// or Binaries > 32 bytes
+func $contract_event(param $topics i64)(param $val i64) (result i64)
 
 /// Internal host function. Same as $system_event but will use ContractEventType::SYSTEM instead
-func $system_event(param $val i64) (result i64)
+func $system_event(param $topics i64)(param $val i64) (result i64)
```
Comment (leighmcculloch) [inline core/cap-0056.md:61]: The term topics is better 👍. Could we rename val to event, or message, to give it more color similar to how we're naming topic?

Comment (leighmcculloch) [reply core/cap-0056.md:61]: Or make it consistent with `data` that is further down.

Comment (sisuresh) [reply core/cap-0056.md]: I renamed `val` to `event`

Comment (leighmcculloch) [reply core/cap-0056.md]: Should the `data` be renamed to `event` further down too?

Comment (sisuresh) [reply core/cap-0056.md]: Sorry, my comment was wrong. I actually renamed `val` to `data`.

**Review by leighmcculloch — APPROVED** [review]

File: core/cap-0056.md (hunk 1 above)
Comment (leighmcculloch) [reply core/cap-0056.md]: Love it. 👍🏻

**Pull request merged**